src/merge_round_robin.rs

```diff
@@ -0,0 +1,150 @@
+use core::num::NonZeroUsize;
+use core::pin::Pin;
+use core::task::Context;
+use core::task::Poll;
+
+use futures::ready;
+use futures::stream::FusedStream;
+use futures::Stream;
+use pin_project_lite::pin_project;
+
+pin_project! {
+    /// Stream for the [`merge_round_robin`](crate::Streamies::merge_round_robin) method.
+    #[derive(Debug)]
+    #[must_use = "streams do nothing unless polled"]
+    pub struct MergeRoundRobin<St1, St2> {
+        #[pin]
+        first: Option<St1>,
+        #[pin]
+        second: Option<St2>,
+
+        first_nb_ele: NonZeroUsize,
+        second_nb_ele: NonZeroUsize,
+
+        first_count: usize,
+        second_count: usize
+    }
+}
+
+impl<St1, St2> MergeRoundRobin<St1, St2>
+where
+    St1: Stream,
+    St2: Stream<Item = St1::Item>,
+{
+    pub(super) fn new(
+        stream1: St1,
+        stream2: St2,
+        first_nb_ele: usize,
+        second_nb_ele: usize,
+    ) -> Self {
+        Self {
+            first: Some(stream1),
+            second: Some(stream2),
+            first_nb_ele: NonZeroUsize::new(first_nb_ele).expect(
+                "Couldn't convert `first_nb_ele` to `NonZeroUsize`. The value must no be 0",
+            ),
+            second_nb_ele: NonZeroUsize::new(second_nb_ele).expect(
+                "Couldn't convert `second_nb_ele` to `NonZeroUsize`. The value must no be 0",
+            ),
+            first_count: 0,
+            second_count: 0,
+        }
+    }
+}
+
+impl<St1, St2> FusedStream for MergeRoundRobin<St1, St2>
+where
+    St1: FusedStream,
+    St2: FusedStream<Item = St1::Item>,
+{
+    fn is_terminated(&self) -> bool {
+        self.first.as_ref().is_none_or(|s| s.is_terminated())
+            && self.second.as_ref().is_none_or(|s| s.is_terminated())
+    }
+}
+
+impl<St1, St2> Stream for MergeRoundRobin<St1, St2>
+where
+    St1: Stream,
+    St2: Stream<Item = St1::Item>,
+{
+    type Item = St1::Item;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        let mut this = self.project();
+        // Check if second stream has finished it's turn
+        if this.second_count == &mut this.second_nb_ele.get() {
+            // It finished. Let's reset the turns
+            *this.first_count = 0;
+            *this.second_count = 0;
+        }
+
+        // Check if we should be polling from the first stream.
+        // This means:
+        //     - It's our turn to be polled AND the se
+        //     - The stream isn't ended
+        if this.first_count < &mut this.first_nb_ele.get() {
+            if let Some(first) = this.first.as_mut().as_pin_mut() {
+                if let Some(item) = ready!(first.poll_next(cx)) {
```

Comment:
Is this behaviour intentional - to wait for the first stream to yield, even if there are items available on the second stream?

My intuition would have been to fall back to the other stream if the first is pending, but I guess it depends on the use case.

Could possibly make this parameterisable or have a different variant which has the fallback behaviour.

Reply:
@RustyNova016 any thoughts on this? I think it'd be nice to have two variants: `merge_round_robin` and `merge_round_robin_strict`.